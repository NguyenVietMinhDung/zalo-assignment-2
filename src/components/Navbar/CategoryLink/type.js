import type { Item } from '../type';

export type Ref = {
  current: any
};

export type Props = {
  ...Item,
  className: ?string
};
