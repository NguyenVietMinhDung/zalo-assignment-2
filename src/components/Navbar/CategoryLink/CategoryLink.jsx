// @flow
import React from 'react';
import './CategoryLink.scss';
import { Props, Ref } from './type';

class CategoryLink extends React.Component<Props> {
  ref: Ref;

  constructor(props: Props) {
    super(props);
    this.ref = React.createRef();
  }

  componentDidUpdate(prevProps: Props) {
    const { className } = this.props;
    if (prevProps.className !== className && className === 'is-current') {
      this.ref.current.scrollIntoView({ inline: 'center' });
    }
  }

  render() {
    const { id, name, className } = this.props;
    return (
      <li className={`category-link ${className}`} ref={this.ref}>
        <a href={`#${id}`}>{name}</a>
      </li>
    );
  }
}

export default CategoryLink;
