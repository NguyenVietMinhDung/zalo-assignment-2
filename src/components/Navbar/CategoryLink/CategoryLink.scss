@import '../../../styles/variables';

.category-link {
	position: relative;

	display: flex;
	flex: 1;

	padding: 0 15px;

	text-align: center;
	align-items: center;
	justify-content: center;

	a {
		color: $light-pink;
	}

	&:hover {
		cursor: pointer;

		a {
			color: white;
		}
	}
}
