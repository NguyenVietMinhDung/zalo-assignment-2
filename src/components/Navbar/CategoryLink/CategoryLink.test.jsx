import React from 'react';
import { shallow } from 'enzyme';
import CategoryLink from './CategoryLink';

const defaultProps = {
  id: 'Other',
  name: 'Kh√°c',
};

describe('<CategoryLink />', () => {
  test('renders correctly', () => {
    const component = shallow(<CategoryLink {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
