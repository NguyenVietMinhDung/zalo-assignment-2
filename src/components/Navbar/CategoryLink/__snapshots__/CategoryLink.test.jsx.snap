// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryLink /> renders correctly 1`] = `
<li
  className="category-link undefined"
>
  <a
    href="#Other"
  >
    Kh√°c
  </a>
</li>
`;
